### PR TITLE
Don't put NULL into strtolower() and preg_match()

### PR DESCRIPTION
**Pull request by alexpott**
https://3v4l.org/KSvQ1/rfc#vgit.master

### CODE REVIEW

**Comment by aik099** [comment]
@alexpott in which PHP version are you getting the deprecation notice shown on https://3v4l.org/KSvQ1/rfc#vgit.master ?

**Comment by aik099** [comment]
Related to https://github.com/minkphp/MinkSelenium2Driver/pull/329 .

**Comment by alexpott** [comment]
PHP 8.1

File: src/WebAssert.php [hunk 1]
```diff
@@ -219,7 +219,7 @@ public function responseHeaderNotContains($name, $value)
      */
     public function responseHeaderMatches($name, $regex)
     {
-        $actual = $this->session->getResponseHeader($name);
+        $actual = (string) $this->session->getResponseHeader($name);
```
Comment (aik099) [inline src/WebAssert.php:222]: Why this (and similar type-casted places) is needed?

**Review by aik099 — CHANGES_REQUESTED** [review]

File: src/WebAssert.php (hunk 1 above)
Comment (alexpott) [reply src/WebAssert.php:222]: https://3v4l.org/hOfGr/rfc#vgit.master

https://php.watch/versions/8.1/internal-func-non-nullable-null-deprecation

**Comment by aik099** [comment]
Merging, thank you @alexpott .

**Pull request merged**